pkg/packetfilter/nftables/namedset.go

```diff
@@ -0,0 +1,142 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nftables
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner/pkg/packetfilter"
+	"sigs.k8s.io/knftables"
+)
+
+type namedSet struct {
+	set      knftables.Set
+	nftables knftables.Interface
+}
+
+func (p *packetFilter) NewNamedSet(set *packetfilter.SetInfo) packetfilter.NamedSet {
+	setType := "ipv4_addr"
+
+	return &namedSet{
+		set: knftables.Set{
+			Name: set.Name,
+			Type: setType,
+		},
+		nftables: p.nftables,
+	}
+}
+
+func (n *namedSet) Create(_ bool) error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Add(&n.set)
+	err := n.nftables.Run(context.TODO(), tx)
+
+	return errors.Wrap(err, "error creating set")
+}
+
+func (n *namedSet) Name() string {
+	return n.set.Name
+}
+
+func (n *namedSet) Flush() error {
+	tx := n.nftables.NewTransaction()
+
+	tx.Flush(&n.set)
+	err := n.nftables.Run(context.TODO(), tx)
```

Comment:
We should check for not found to be consistent with iptables.

```suggestion
	err := n.nftables.Run(context.TODO(), tx)
	if knftables.IsNotFound(err) {
		return nil
	}
```